lib/Transforms/HC/SelectAcceleratorCode/SelectAcceleratorCode.cpp

```diff
@@ -0,0 +1,175 @@
+//===------ SelectAcceleratorCode.cpp - Select only accelerator code ------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+//
+// This file declares and defines a pass which selects only code which is
+// expected to be run by an accelerator i.e. referenced directly or indirectly
+// (through a fully inlineable call-chain) by a [[hc]] function. To support
+// subsequent processing, it also marks all identified functions as AlwaysInline
+// thus making it possible to use only the AlwaysInliner without resorting to a
+// more expensive full Inliner pass.
```

Comment:
@AlexVlx thanks for putting comments in this pass. It answers one question i had in the PR for `hcc`.